visualizer/actionButton.js

```diff
@@ -0,0 +1,265 @@
+/* eslint-env browser */
+
+'use strict';
+
+(function(root, initModule) {
+  if (typeof exports === 'object') {
+    module.exports = initModule;
+  } else {
+    initModule(root.ohmEditor);
+  }
+})(this, function(ohmEditor) {
+  var rootTrace = null;
+  function rebootRootTraceForSemantics(matchResult, trace) {
+    rootTrace = trace;
+  }
+  ohmEditor.addListener('parse:input', rebootRootTraceForSemantics);
+
+  var UnicodeChars = {
+    BLACK_UP_POINTING_TRIANGLE: '\u25B2',
+    BLACK_DOWN_POINTING_TRIANGLE: '\u25BC',
+    PLUS_SIGN: '\u002B',
+    LEFTWARDS_ARROW: '\u2190'
+  };
+
+  function createActionNameContainer() {
```

Comment:
I think everything you're calling "action" in this file is really an "operation", right? I think the terminology we've been using is that "semantic action" means a function for a specific node type, and "operation" is `toAST()`, `value`, etc.